src/devices/Pigpio/Driver.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Pigpio
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class Driver : GpioDriver
+    {
+        private readonly IPEndPoint _endpoint;
+        private readonly PigpiodIf _proxy;
+        private readonly List<int> _openPins;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="endpoint"></param>
+        public Driver(IPEndPoint endpoint)
+        {
+            _endpoint = endpoint;
+            _proxy = new PigpiodIf();
+
+            _openPins = new List<int>();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        public Task ConnectAsync()
+        {
+            _proxy.pigpio_start(_endpoint.Address.ToString(), _endpoint.Port.ToString());
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc/>
+        protected override int PinCount => PigpiodIf.PI_MAX_USER_GPIO;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="pinNumber"></param>
+        /// <param name="eventTypes"></param>
+        /// <param name="callback"></param>
+        protected override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventTypes, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        protected override void ClosePin(int pinNumber)
+        {
+            if (_openPins.Contains(pinNumber))
```

Comment:
this is linear operation on the list. With HashSet it's amortized `O(log(n))` with median `O(1)`